frontend/src/plugin/registry.tsx

```diff
@@ -758,6 +771,147 @@ export function registerKindIcon(kind: string, definition: IconDefinition) {
   store.dispatch(graphViewSlice.actions.addKindIcon({ kind, definition }));
 }
 
+/**
+ * Register a new cluster action menu item.
+ * @param item - The item to add to the cluster action menu.
+ *
+ * @example
+ *
+ * ```tsx
+ * import { registerClusterProviderMenuItem } from '@kinvolk/headlamp-plugin/lib';
+ * import { MenuItem, ListItemText } from '@mui/material';
+ * registerClusterProviderMenuItem(({cluster, setOpenConfirmDialog, handleMenuClose}) => {
+ *  const isMinikube =
+ *   cluster.meta_data?.extensions?.context_info?.provider === 'minikube.sigs.k8s.io';
+ *   if (!helpers.isElectron() !! !isMinikube) {
+ *     return null;
+ *   }
+ *   return (
+ *     <MenuItem
+ *       onClick={() => {
+ *        setOpenConfirmDialog('deleteMinikube');
+ *        handleMenuClose();
+ *       }}
+ *     >
+ *       <ListItemText>{t('translation|Delete')}</ListItemText>
+ *     </MenuItem>
+ *   );
+ * )}
+ * ```
+ *
+ */
+export function registerClusterProviderMenuItem(item: MenuItemComponent) {
+  store.dispatch(addMenuItem(item));
+}
+
+/**
+ * Register a new cluster provider dialog.
```

Comment:
I'm not sure I understand what "cluster provider dialog" means from just this comment and example